### PR TITLE
LibPDF: Implement rest of AESV3 support :^)

### PR DESCRIPTION
**Pull request by nico**
Demo:

```sh
% Build/lagom/bin/pdf /Users/thakis/Downloads/CIPA_DC-007-2021_E.pdf
PDF Version: 1.7
Number of pages: 63
Title: CIPA DC-007-Translation-2021 Multi-Picture Format.
Author: CIPA
Subject: 
Keywords: 
Creator: Word 用 Acrobat PDFMaker 17
Producer: Adobe PDF Library 17.11.238
Creation date: D:20211224140020+09'00'
Modification date: D:20211224140459+09'00'
```

"LibPDF: Implement 7.6.4.3.3 Algorithm 2.A: Retrieve file encryption key" is the most interesting commit. The commit message is fairly detailed. The rest is fairly straightforward. (That commit is also fairly straightforward, but a bit longer.)

### CODE REVIEW

**Comment by nico** [comment]
I suppose "LibPDF: Implement 7.6.4.4.11 Algorithm 12: Authenticating owner password" is also a bit interesting since it contains code to work around non-spec-compliant files.

**Comment by nico** [comment]
Paging folks who've reviewed earlier patches on this: @mattco98 @alimpfard @linusg

**Review by mattco98 — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibPDF/Encryption.cpp [hunk 1]
```diff
@@ -286,16 +313,41 @@ bool StandardSecurityHandler::authenticate_user_password_r6_and_later(StringView
...
 
 bool StandardSecurityHandler::try_provide_user_password(StringView password_string)
 {
     bool has_user_password;
-    if (m_revision >= 6)
-        has_user_password = authenticate_user_password_r6_and_later(password_string);
-    else
+    if (m_revision >= 6) {
+        // This checks both owner and user password.
+        has_user_password = compute_encryption_key_r6_and_later(MUST(ByteBuffer::copy(password_string.bytes())));
```
[the remark this answers is not in the document]
Comment (nico) [reply Userland/Libraries/LibPDF/Encryption.cpp]: Meh. Done, but feels fairly unrelated to most of this PR.

**Review by mattco98 — APPROVED** [review]

**Comment by nico** [comment]
`builtins/Temporal/Now/Now.plainDateTimeISO.js` failure on `SerenityOS Serenity_Clang_x86_64_Coverage_OFF` is unrelated.

**Pull request merged**